docs/generators.md

```diff
@@ -0,0 +1,53 @@
+# Generators
+
+**Generators** are classes that are used to render models for a given language.
+
+## Supported languages
+
+- [JavaScript](../src/generators/javascript/JavaScriptGenerator.ts),
+- [TypeScript](../src/generators/typescript/TypeScriptGenerator.ts),
+- [Java](../src/generators/java/JavaGenerator.ts).
+
+## Generator's options
+
+Each generator extends default options for generator. It means that generator can also have additional options.
+
+Options is passed as first argument to generator's constructor. Check the example:
+
+```ts
+const generator = new TypeScriptGenerator({ ...options });
+```
+
+Default options contains:
+
+| Option | Type | Description | Default value |
+|---|---|---|---|---|
+| `indentation` | Object | Options for indentation. | - |
+| `indentation.type` | String | Type of indentation. Its value can be either `SPACES` or `TABS`. | `SPACES` |
+| `indentation.size` | String | Size of indentation. | 2 |
+| `defaultPreset` | Object | Default preset for generator. For more information, read [customisation](./customisation.md) document. | _Implemented by generator_ |
+| `presets` | Array | Array contains **presets**. For more information, read [customisation](./customisation.md) document. | `[]` |
+
+Below is a list with additional options available for a given generator.
+
+### [TypeScript](../src/generators/typescript/TypeScriptGenerator.ts)
+
+| Option | Type | Description | Default value |
+|---|---|---|---|---|
+| `renderTypes` | Boolean | Renders or not signature for types. | `true` |
+| `modelType` | String | It indicates which model type should be rendered for `object` type. Its value can be either `interface` or `class`. | `class` |
+
```

Comment:
The table is not showing up properly when rendered - https://github.com/asyncapi/generator-model-sdk/blob/a5817b374172a673f5dd567d71823d741eb997f4/docs/generators.md